src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -205,20 +205,27 @@ public static ClassName get(String packageName, String simpleName, String... sim
   public static ClassName get(TypeElement element) {
     checkNotNull(element, "element == null");
     List<String> names = new ArrayList<>();
-    for (Element e = element; isClassOrInterface(e); e = e.getEnclosingElement()) {
-      checkArgument(element.getNestingKind() == TOP_LEVEL || element.getNestingKind() == MEMBER,
+    TypeElement e = element;
+    // Add nested class names one at a time before adding the package.
```

Comment:
the comment says what it does. Can you add why?

Reply:
Sure.  I'm not 100% sure why (the old code was already doing this), but I think it's just "`because {@link #names} expects the package first as a single item.`"